Add name_scope to HyperParameters and HyperResnet to applications

omalleyt12 opened the pull request:
Put all of the logic for HyperResnet into one file to remove abstraction, and used the `name_scope` ability to simplify defining conditional hyperparameters

fchollet commented on kerastuner/applications/resnet.py:
```diff
@@ -12,14 +12,106 @@
+        # Version-conditional fixed parameters
+        preact = True if version == 'v2' else False
+        use_bias = False if version == 'next' else True
+
```
Remove blank line

omalleyt12 replied:
Done

fchollet commented on kerastuner/engine/hyperparameters.py:
```diff
@@ -205,13 +206,23 @@ class HyperParameters(object):
     def __init__(self):
         self.space = []
         self.values = {}
+        self.name_scopes = []
```
This should be private

omalleyt12 replied:
Done

fchollet commented on tests/kerastuner/applications/resnet_test.py:
```diff
@@ -0,0 +1,58 @@
+
```
Remove blank line

omalleyt12 replied:
Done

fchollet commented on kerastuner/applications/resnet.py:
```diff
@@ -12,14 +12,106 @@
+        x = layers.ZeroPadding2D(padding=((3, 3), (3, 3)), name='conv1_pad')(x)
+        x = layers.Conv2D(
+            64,
+            7,
```
Style nit: you can put all unnamed positional args on one line

omalleyt12 replied:
Done

fchollet commented on tests/kerastuner/applications/resnet_test.py:
```diff
@@ -0,0 +1,58 @@
+def test_model_construction():
+    hp = hp_module.HyperParameters()
+    hypermodel = resnet.HyperResnet(input_shape=(256, 256, 3), num_classes=10)
+    model = hypermodel.build(hp)
```
We should run basic tests on the model to check it is a valid model with the expected behavior (run it on a test input to check output shape; try train_on_batch to test compile config...)

For best coverage, we should do such checks for every possible configuration (albeit that may be challenging), otherwise we can't be sure there aren't invalid configurations in the mix.

omalleyt12 replied:
Done, added tests parametrized by the version, this test is now running in ~230 sec on my laptop so I wasn't sure if I should add more?

fchollet replied:
Test are stalling unfortunately: https://travis-ci.org/keras-team/keras-tuner/jobs/547946699

Let's only run the top 5 smallest models on CI, and configure the test to run all models when run locally.

fchollet commented on tests/kerastuner/applications/resnet_test.py:
```diff
@@ -0,0 +1,65 @@
+    assert model.layers
+    assert model.name == 'Resnet'
+    assert model.output_shape == (None, 10)
+    model.train_on_batch(np.ones((10, 256, 256, 3)), np.ones((10, 10)))
```
You'll go faster with 1 sample per batch (same below)

omalleyt12 replied:
Done

omalleyt12 commented:
Sounds good, how do I configure a test to run differently locally vs on Travis?

fchollet commented:
I'd suggest editing travis.yml to add a new APPLICATIONS flag, to get your own Travis build (later on we will skip running this build if the applications file are left unedited). Then check for this flag in the code, and skip tests or not based on the flag.

omalleyt12 commented:
Added HyperXception and added an env variable to skip large tests in TRAVIS. Never used Travis CI before, searched around the docs a bit and came up with this but let me know if this isn't what you meant

fchollet commented on kerastuner/applications/xception.py:
```diff
@@ -12,10 +12,91 @@
 from tensorflow.keras import layers
-from tensorflow.keras.layers import *
+from tensorflow.keras.callbacks import EarlyStopping, ModelCheckpoint
+from tensorflow.keras.utils import to_categorical
```
Unused?

omalleyt12 replied:
Ah, yep removed

fchollet commented on kerastuner/applications/xception.py:
```diff
@@ -12,10 +12,91 @@
 import tensorflow.keras as keras
 from tensorflow.keras import layers
-from tensorflow.keras.layers import *
+from tensorflow.keras.callbacks import EarlyStopping, ModelCheckpoint
```
Unused?

omalleyt12 replied:
Done

fchollet commented on kerastuner/applications/xception.py:
```diff
@@ -12,10 +12,91 @@
 # limitations under the License.
 
+import copy
+import functools
```
Unused?

omalleyt12 replied:
Done

omalleyt12 commented:
Thanks for the review!
